flink-test-utils-parent/flink-connector-test-utils/src/main/java/org/apache/flink/connector/testutils/source/reader/SourceReaderTestBase.java

```diff
@@ -207,19 +207,16 @@ public void collect(Integer element, long timestamp) {
 
         public void validate() {
 
-            assertEquals(
-                    String.format("Should be %d distinct elements in total", totalNumRecords),
-                    totalNumRecords,
-                    consumedValues.size());
-            assertEquals(
-                    String.format("Should be %d elements in total", totalNumRecords),
-                    totalNumRecords,
-                    count);
-            assertEquals("The min value should be 0", 0, min);
-            assertEquals(
-                    String.format("The max value should be %d", totalNumRecords - 1),
-                    totalNumRecords - 1,
-                    max);
+            assertThat(consumedValues)
+                    .withFailMessage("Should be %d distinct elements in total", totalNumRecords)
```

Comment:
Use `as()` instead

Reply:
The text used in the old JUnit4 should be more like an error msg than describing the test. It should be better to use withFailMessage(..) in this case.  @PatrickRen WDYT? 

Reference: https://junit.org/junit4/javadoc/4.8/org/junit/Assert.html#assertEquals(java.lang.String,%20long,%20long). The message is used for AssertionError.